spec/api_client.js

```diff
@@ -8,29 +8,37 @@ describe('api client', function () {
 
   it('should make a get request with correct jwt token', function (done) {
 
-    var urlBase = 'http://base',
+    var urlBase = 'https://api.notifications.service.gov.uk',
       path = '/email',
       body = {
         'body': 'body text'
       },
-      serviceId = 123,
-      apiKeyId = 'SECRET',
-      apiClient = new ApiClient(urlBase, serviceId, apiKeyId);
+      serviceId = 'c745a8d8-b48a-4b0d-96e5-dbea0165ebd1',
+      apiKeyId = '8b3aa916-ec82-434e-b0c5-d5d9b371d6a3';
 
-    nock(urlBase, {
-      reqheaders: {
-        'Authorization': 'Bearer ' + createGovukNotifyToken('GET', path, apiKeyId, serviceId)
-      }
-    })
-      .get(path)
-      .reply(200, body);
+    [
+      new ApiClient(serviceId, apiKeyId),
+      new ApiClient(urlBase, serviceId, apiKeyId),
+      new ApiClient(urlBase, 'key_name' + '-' + serviceId + '-' + apiKeyId),
+      new ApiClient('key_name' + ':' + serviceId + ':' + apiKeyId),
+    ].forEach(function(client, index, clients) {
+
+      nock(urlBase, {
+        reqheaders: {
+          'Authorization': 'Bearer ' + createGovukNotifyToken('GET', path, apiKeyId, serviceId)
+        }
+      })
+        .get(path)
+        .reply(200, body);
```

Comment:
not familiar with nock, but can we do any more assertions here to make sure that we're actually calling `createToken` with the correct thing? since that's the thing i'm concerned about here